v1/ast/policy.go

```diff
@@ -8,21 +8,13 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
-	"math/rand"
 	"strings"
-	"time"
 
 	"github.com/open-policy-agent/opa/v1/ast/internal/tokens"
 	astJSON "github.com/open-policy-agent/opa/v1/ast/json"
 	"github.com/open-policy-agent/opa/v1/util"
 )
 
-// Initialize seed for term hashing. This is intentionally placed before the
-// root document sets are constructed to ensure they use the same hash seed as
-// subsequent lookups. If the hash seeds are out of sync, lookups will fail.
-var hashSeed = rand.New(rand.NewSource(time.Now().UnixNano()))
-var hashSeed0 = (uint64(hashSeed.Uint32()) << 32) | uint64(hashSeed.Uint32())
```

Comment:
After running `git blame`, this looks like leftover code to me.  

7 years ago, we switched from SipHash to xxHash in PR https://github.com/open-policy-agent/opa/pull/970. I'm not entirely sure why the author of that PR removed the usage of the seed for `(Number).Hash` but kept it for `(String).Hash` and `(Var).Hash` in commit 5cb4d8fd11060e078699efd0aeae412559e3613d.

Reply:
I don't know that, but I'd rather have it added back for number than removed from here. Or why would we not include a seed in the hashes?

Reply:
> Or why would we not include a seed in the hashes?

My guess is that the initial implementation used the SipHash package, which requires two seeds: https://pkg.go.dev/github.com/dchest/siphash#Hash

```
func Hash(k0, k1 uint64, b []byte) uint64
```

Since the xxHash package we use now doesn’t enforce a seed, I think we don’t need them anymore. Also,

```go
var hashSeed = rand.New(rand.NewSource(time.Now().UnixNano()))
```

relies on `time.Now().UnixNano()`, so it’s not a constant seed either.

Reply:
Thanks! I haven't really dug into the details here, so somebody else will be a better judge here. 

Thanks a lot for submitting this though! Seems like a good and relatively simple improvement to me.

Reply:
So in the used xxhash (cespare) it's defaulting to hashseed = 0 if unset. https://github.com/cespare/xxhash/blob/main/xxhash.go#L39-L42

What are our hashes for? I don't think we'll need to worry about them being predictable, other than that some iteration order might become more stable than advertised... 🤔 @johanfylling do you see a need for this?

Reply:
Thanks!